pkg/scheduler/framework/plugins/podtopologyspread/pod_topology_spread.go

```diff
@@ -51,7 +110,172 @@ func (pl *PodTopologySpread) Filter(ctx context.Context, cycleState *framework.C
 	return migration.PredicateResultToFrameworkStatus(reasons, err)
 }
 
+// PostFilter invoked at the postFilter extension point.
+// Here it prepares necessary data (podTopologySpreadMap) for incoming pod on the filteredNodes.
+// And later Score plugin is able to use 'podTopologySpreadMap' to perform the Score/Normalize calculations.
+func (pl *PodTopologySpread) PostFilter(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, filteredNodes []*v1.Node, _ framework.NodeToStatusMap) *framework.Status {
```

Comment:
I think we shouldn't add a PostFilter just yet, for phase 1 of the migration, we don't want to duplicate any logic, we just want to wrap the existing logic around Plugins.

What I think we should do is the following:
1) This PostFilter logic should go to PriorityMetadata
2) Update the existing priority function to execute as priority map/reduce that consumes this metadata (basically the Score and NormalizeScore logic you have here move them to map/reduce)
3) The Score function here just calls the map function, while the Normalize just calls the NormalizeScore function.